internal/dnstransport/dnsoverudp/dnsoverudp_test.go

```diff
@@ -10,31 +9,44 @@ import (
 	"github.com/miekg/dns"
 	"github.com/ooni/netx/handlers"
 	"github.com/ooni/netx/internal/connx"
-	"github.com/ooni/netx/internal/dnstransport/dnsoverudp"
 )
 
 func TestIntegrationSuccess(t *testing.T) {
-	transport := dnsoverudp.NewTransport(
-		time.Now(), handlers.NoHandler, "9.9.9.9:53",
+	transport := NewTransport(
+		net.Dial, "9.9.9.9:53",
```

Comment:
test failed: 
```
$ go test -v  ./internal/dnstransport/dnsoverudp/
=== RUN   TestIntegrationSuccess
--- FAIL: TestIntegrationSuccess (3.00s)
    dnsoverudp_test.go:20: read udp 192.168.128.16:41657->9.9.9.9:53: i/o timeout
...
```
but `dnsclient` is fine :
```
$ ./dnsclient -transport udp -endpoint 9.9.9.9:53 -name google.com
...
["172.217.17.46","2a00:1450:400e:808::200e"]
null
```

Reply:
I believe this is a transient error that we can ignore for now

Reply:
The "problem", if you wish, is this code:

```Go
	err = conn.SetDeadline(time.Now().Add(3 * time.Second))
	if err != nil {
		return
	}
```

I'm letting the UDP dialer fail with timeout after 3 seconds.

Reply:
Issue mentioning calibration: https://github.com/ooni/netx/issues/29

Reply:
Thanks.